### PR TITLE
Update redgate step-templates

### PR DESCRIPTION
**Pull request by Rixium**
# Step template checklist

- [x] `Id` should be a **GUID** that is not `00000000-0000-0000-0000-000000000000`
- [x] `Version` should be incremented, otherwise the integration with Octopus won't update the step template correctly
- [x] Parameter names should not start with `$`
- [x] **Step template parameter names (the ones declared in the JSON, not the script body) should be prefixed with a namespace so that they are less likely to clash with other user-defined variables in Octopus** (see [this issue](https://github.com/OctopusDeploy/Issues/issues/2126)). For example, use an abbreviated name of the step template or the category of the step template).
- [x] `LastModifiedBy` field must be present, and (_optionally_) updated with the correct author

### CODE REVIEW

File: step-templates/redgate-deploy-from-package.json [hunk 1]
```diff
@@ -122,16 +122,25 @@
                                                }
                        },
                        {
-                           "Name":  "SpecificModuleVersion",
+                           "Name":  "DLMAutomationSpecificModuleVersion",
```
Comment (OctopusBob) [inline step-templates/redgate-deploy-from-package.json:125]: I understand the desire to want to change this variable name to be consistent, but anyone using this step will be forced to update this value (if they have it set).  Or, they could lose the value they have set already.  Is it absolutely required to change this variable name?

image

File: step-templates/redgate-deploy-from-database.json [hunk 2]
```diff
@@ -140,16 +140,25 @@
                                                }
                        },
                        {
-                           "Name":  "SpecificModuleVersion",
+                           "Name":  "DLMAutomationSpecificModuleVersion",
```
Comment (OctopusBob) [inline step-templates/redgate-deploy-from-database.json:143]: I understand the desire to want to change this variable name to be consistent, but anyone using this step will be forced to update this value (if they have it set).  Or, they could lose the value they have set already.  Is it absolutely required to change this variable name?

image

File: step-templates/redgate-deploy-from-database-release.json [hunk 3]
```diff
@@ -85,16 +85,25 @@
                                                }
                        },
                        {
-                           "Name":  "SpecificModuleVersion",
+                           "Name":  "DLMAutomationSpecificModuleVersion",
```
Comment (OctopusBob) [inline step-templates/redgate-deploy-from-database-release.json:88]: I understand the desire to want to change this variable name to be consistent, but anyone using this step will be forced to update this value (if they have it set).  Or, they could lose the value they have set already.  Is it absolutely required to change this variable name?

image

File: step-templates/redgate-create-database-release.json [hunk 4]
```diff
@@ -131,16 +131,25 @@
                                                }
                        },
                        {
-                           "Name":  "SpecificModuleVersion",
+                           "Name":  "DLMAutomationSpecificModuleVersion",
```
Comment (OctopusBob) [inline step-templates/redgate-create-database-release.json:134]: I understand the desire to want to change this variable name to be consistent, but anyone using this step will be forced to update this value (if they have it set).  Or, they could lose the value they have set already.  Is it absolutely required to change this variable name?

image

Comment (Rixium) [reply step-templates/redgate-create-database-release.json:134]: I changed it to match the requirement in step 4 of your checklist, but I can change this back to the previous value if you'd prefer.

Comment (OctopusBob) [reply step-templates/redgate-create-database-release.json:134]: It is a risk/reward concern.  I appreciate you following the guidelines.  However, that parameter was added two years ago and we haven't gotten a complaint about it (unless you have).  I think it is safe to revert back to the previous value in this case.

Comment (Rixium) [reply step-templates/redgate-create-database-release.json:134]: Great, thank you! I'll do that, and update this pull-request when I've done so.

**Comment by Rixium** [comment]
> I noticed a new function `ConfigureProxyIfEnvironmentVariableSet` in these step templates.
> 
> ```
> if (Test-Path env:REDGATE_PROXY_URL) {
>         Write-Debug "Setting DefaultWebProxy to $env:REDGATE_PROXY_URL"
> 
>         [System.Net.WebRequest]::DefaultWebProxy = New-Object System.Net.WebProxy($env:REDGATE_PROXY_URL)
>         [System.Net.WebRequest]::DefaultWebProxy.credentials = [System.Net.CredentialCache]::DefaultNetworkCredentials
>         [System.Net.WebRequest]::DefaultWebProxy.BypassProxyOnLocal = $True
>     }
> ```
> 
> Is that environment variable `env:REDGATE_PROXY_URL` documented anywhere? Would it be possible to make that a parameter as well to make it more discoverable?


I didn't personally make the change, so I've asked for some clarification on that, and will update you asap.

**Comment by Rixium** [comment]
@OctopusBob We've included the requested change, so now the proxy URL can be set by the user in the step-template. I've also reverted the parameter "SpecificModuleVersion"  back to its original name.

**Comment by Rixium** [comment]
We are considering a few things internally with our proxy URL parameter, so we would prefer not to merge this change just yet. I'm going to close the PR until we have decided a way forward.

**Pull request closed without merging**